cdap-ui/app/cdap/components/AbstractWidget/FieldInputDropdown/index.tsx

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+*/
+import React from 'react';
+import PropTypes from 'prop-types';
+import Select from 'components/AbstractWidget/FormInputs/Select';
+
+interface IStage {
+  name: string;
+  schema: string;
+}
+
+interface IField {
+  name: string;
+  type: string;
+}
+
+interface IFieldInputProps {
+  value: string;
+  onChange: (value: string) => void;
+  inputSchema: IStage[];
+  disabled?: boolean;
+}
+
+// We are assuming all incoming stages have the same schema
+function getFields(schemas: IStage[]) {
+  let fields = [];
+  if (schemas.length === 0) {
```

Comment:
I think schemas can be undefined as well, so it's probably good to check existence of schemas first before checking length